src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs

```diff
@@ -290,7 +290,36 @@ protected override async Task<IImmutableList<Exception>> WriteMessagesAsync(IEnu
                 NotifyNewEventAppended();
             return result;
         }
+        private async ValueTask InsertEntries(IEnumerable<JournalEntry> entries)
+        {
+            if (_settings.Transaction)
+            {
+                var sessionOptions = new ClientSessionOptions { };
+                using (var session = await _client.StartSessionAsync(sessionOptions/*, cancellationToken*/))
+                {
+                    // Begin transaction
+                    session.StartTransaction();
+                    try
+                    {
+                        foreach (var entry in entries)
+                        {
+                            await _journalCollection.Value.InsertOneAsync(session, entry);
+                        }
 
+                        //All good , lets commit the transaction 
+                        await session.CommitTransactionAsync();
```

Comment:
This is the wrong place to commit the transaction, it should be outside the try...catch block